[googletest] Add definitions that opt-out of polluting macros

Reasoning for not upstreaming this patch: Drake-specific preference.

--- BUILD.bazel
+++ BUILD.bazel
@@ -113,3 +113,7 @@
     }),
-    defines = select({
+    defines = [
+        "GTEST_DONT_DEFINE_FAIL=1",
+        "GTEST_DONT_DEFINE_SUCCEED=1",
+        "GTEST_DONT_DEFINE_TEST=1",
+    ] + select({
         ":has_absl": ["GTEST_HAS_ABSL=1"],
